File: 0013-roman-to-integer/0013-roman-to-integer.py
class Solution:
    def romanToInt(self, s: str) -> int:
        r_to_i = {'I':1, 'V':5, 'X':10, 'L':50, 'C':100, 'D':500, 'M':1000}
        if len(s) == 1:
            return r_to_i[s]
        rsum = 0
        i = 0
        while i<len(s)-1:
            first, second = r_to_i[s[i]], r_to_i[s[i+1]]
            if first < second:
                rsum += second-first
                i += 1
            else:
                rsum += first
            i += 1
        if i<len(s):
            rsum += r_to_i[s[i]]
        return rsum

File: 0013-roman-to-integer/0013-roman-to-integer.cpp
class Solution {
public:
    int romanToInt(string s) {
        unordered_map<char, int> r_to_i {
            {'I', 1}, {'V', 5}, {'X', 10}, {'L', 50}, 
            {'C', 100}, {'D', 500}, {'M', 1000}
        };
        int value {}, prev {}, cur {}, n { static_cast<int>(s.length()) };
        for (int i { n-1 }; i>=0; i--) {
            cur = r_to_i[s[i]];
            value += cur<prev ? -cur : cur;
            prev = cur;
        }
        return value;
    }
};